### PR TITLE
[serve] migrate serve ha integration test to civ2

### PR DESCRIPTION
**Pull request by aslonnie**
and rename `xcommit` tag to `ha_integration`

### CODE REVIEW

File: .buildkite/serve.rayci.yml [hunk 1]
```diff
@@ -147,6 +147,22 @@ steps:
...
     depends_on: servebuild
 
+  - label: ":ray-serve: serve: HA integration tests"
+    tags:
+      - serve
+      - python
+    instance_type: medium
+    commands:
+      - bazel run //ci/ray_ci:build_in_docker -- docker --python-version 3.8 --platform cpu --canonical-tag ha_integration
+      - docker tag rayproject/ray:ha_integration ray_ci:v1
```
Comment (can-anyscale) [inline .buildkite/serve.rayci.yml:157]: where is the `ray_ci:v1` tag is defined? maybe just us the same name overthere and remove this line?

**Review by can-anyscale — APPROVED** [review]

File: .buildkite/serve.rayci.yml (hunk 1 above)
Comment (aslonnie) [reply .buildkite/serve.rayci.yml:157]: it is in the `python/ray/tests/conftest.py`. I can do it as a follow up (and cleanup the old `prepare_docker()` function entirely.

**Comment by aslonnie** [comment]
@edoakes , before this change, `ml` tag will also trigger serve ha tests. is that really required?

**Comment by aslonnie** [comment]
https://buildkite.com/ray-project/premerge/builds/14038#018c4712-02d2-4c61-9d25-50fde76324ba

**Pull request merged**